static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -0,0 +1,413 @@
+import {getExactDuration, parseLargestSuffix} from 'sentry/utils/formatters';
+import useApi from 'sentry/utils/useApi';
+
+import {Threshold} from '../utils/types';
+
+const NEW_THRESHOLD_PREFIX = 'newthreshold';
+
+type Props = {
+  columns: number;
+  orgSlug: string;
+  refetch: () => void;
+  setError: (msg: string) => void;
+  thresholds: Threshold[];
+};
+
+type EditingThreshold = {
+  environment: Environment;
+  id: string;
+  project: Project;
+  threshold_type: string;
+  trigger_type: string;
+  value: number;
+  windowSuffix: string;
+  windowValue: number;
+  date_added?: string;
+  hasError?: boolean;
+  window_in_seconds?: number;
+};
+
+export function ThresholdGroupRows({
+  thresholds,
+  columns,
+  orgSlug,
+  refetch,
+  setError,
+}: Props) {
+  const [editingThresholds, setEditingThresholds] = useState<{
+    [key: string]: EditingThreshold;
+  }>({});
+  const [newThresholdIterator, setNewThresholdIterator] = useState<number>(0); // used simply to initialize new threshold
+  const api = useApi();
+  const project = thresholds[0].project;
+  const environment = thresholds[0].environment;
+  const defaultWindow = thresholds[0].window_in_seconds;
+
+  const thresholdsById: {[id: string]: Threshold} = useMemo(() => {
+    const byId = {};
+    thresholds.forEach(threshold => {
+      byId[threshold.id] = threshold;
+    });
+    return byId;
+  }, [thresholds]);
+
+  const thresholdIdSet = useMemo(() => {
+    return new Set([
+      ...thresholds.map(threshold => threshold.id),
+      ...Object.keys(editingThresholds),
+    ]);
+  }, [thresholds, editingThresholds]);
+
+  const initializeNewThreshold = () => {
+    const thresholdId = `${NEW_THRESHOLD_PREFIX}-${newThresholdIterator}`;
+    const [windowValue, windowSuffix] = parseLargestSuffix(defaultWindow);
+    const newThreshold: EditingThreshold = {
+      id: thresholdId,
+      project,
+      environment,
+      windowValue,
+      windowSuffix,
+      threshold_type: 'total_error_count',
+      trigger_type: 'over',
+      value: 0,
+      hasError: false,
+    };
+    const updatedEditingThresholds = {...editingThresholds};
+    updatedEditingThresholds[thresholdId] = newThreshold;
+    setEditingThresholds(updatedEditingThresholds);
+    setNewThresholdIterator(newThresholdIterator + 1);
+  };
+
+  const enableEditThreshold = thresholdId => {
+    const updatedEditingThresholds = {...editingThresholds};
+    const threshold = JSON.parse(JSON.stringify(thresholdsById[thresholdId]));
+    const [windowValue, windowSuffix] = parseLargestSuffix(threshold.window_in_seconds);
+    updatedEditingThresholds[thresholdId] = {
+      ...threshold,
+      windowValue,
+      windowSuffix,
+      hasError: false,
+    };
+    setEditingThresholds(updatedEditingThresholds);
+  };
+
+  const saveThreshold = thresholdIds => {
+    thresholdIds.forEach(id => {
+      const thresholdData = editingThresholds[id];
+      const submitData: {[key: string]: any} = {...thresholdData};
```

Comment:
maybe something like this would avoid casting it to any type
```suggestion
      const submitData= {
      	...thresholdData,
      window_in_seconds: seconds,
      environment: thresholdData.environment.name; // api expects environment as a string
      };
```